.github/workflows/upgrade-python-patch-version.yml

```diff
@@ -0,0 +1,95 @@
+name: Upgrade Python patch version
+
+on:
+  workflow_dispatch:
+  schedule:
+    # At 4AM on Tuesday (UTC) - runs after dependency updates on Monday
+    - cron: "0 4 * * 2"
+
+permissions: {}
+
+jobs:
+  upgrade_python_version:
+    name: Upgrade Python patch version
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # This is required for getting the required OIDC token from GitHub
+    environment:
+      name: main
+    steps:
+      - uses: DataDog/dd-octo-sts-action@acaa02eee7e3bb0839e4272dacb37b8f3b58ba80 # v1.0.3
+        id: octo-sts
+        with:
+          scope: DataDog/datadog-agent
+          policy: self.update-dependencies.create-pr
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          # credentials are needed to create the PR at the end of the workflow
+          persist-credentials: true
+
+      - name: Set up Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: '3.13'
+      - name: Install dda
+        uses: ./.github/actions/install-dda
+        with:
+          features: legacy-tasks
+      - name: Install Python dependencies
+        run: pip install packaging httpx orjson
+
+      - name: Update Python version
+        id: update
+        run: |
+          # Get current version before update
+          CURRENT_VERSION=$(dda inv python-version.get)
```

Comment:
Can these dda commands fail by any chance?

Reply:
Can you elaborate?

Reply:
Okay